python-avd/pyavd/_eos_designs/structured_config/flows/__init__.py

```diff
@@ -124,70 +109,96 @@ def _enable_sflow(self) -> bool:
-        flow_tracking[tracker_type]["shutdown"] = False
+        self.structured_config.flow_tracking.hardware.record.format_ipfix_standard_timestamps_counters = (
+            self.inputs.flow_tracking_settings.hardware.record.format_ipfix_standard_timestamps_counters
+        )
+        self.structured_config.flow_tracking.hardware.shutdown = False
+
+        # Validate and configure trackers
+        for tracker_name in natural_sort(trackers):
+            config = self._get_tracker_input_config(tracker_name)
+            self.structured_config.flow_tracking.hardware.trackers.append_new(
+                name=config.name,
+                record_export=config.record_export._cast_as(EosCliConfigGen.FlowTracking.Hardware.TrackersItem.RecordExport),
+                exporters=config.exporters._cast_as(EosCliConfigGen.FlowTracking.Hardware.TrackersItem.Exporters),
+            )
+
+    def _set_sampled_flow_tracking(self) -> None:
+        """Set the structured configuration for sampled flow tracking if any interface is configured."""
+        all_interfaces = chain(
+            self.structured_config.ethernet_interfaces, self.structured_config.port_channel_interfaces, self.structured_config.dps_interfaces
+        )
+
+        trackers: set[str] = {interface.flow_tracker.sampled for interface in all_interfaces if interface.flow_tracker.sampled}
+        if not trackers:
+            return
+
+        self.structured_config.flow_tracking.sampled._update(
+            sample=self.inputs.flow_tracking_settings.sampled.sample,
+            shutdown=False,
+            encapsulation=EosCliConfigGen.FlowTracking.Sampled.Encapsulation(
+                ipv4_ipv6=self.inputs.flow_tracking_settings.sampled.encapsulation.ipv4_ipv6,
+                mpls=self.inputs.flow_tracking_settings.sampled.encapsulation.mpls,
+            ),
+            hardware_offload=EosCliConfigGen.FlowTracking.Sampled.HardwareOffload(
+                ipv4=self.inputs.flow_tracking_settings.sampled.hardware_offload.ipv4,
+                ipv6=self.inputs.flow_tracking_settings.sampled.hardware_offload.ipv6,
+                threshold_minimum=self.inputs.flow_tracking_settings.sampled.hardware_offload.threshold_minimum,
+            ),
+        )
+
+        # Validate and configure trackers
+        for tracker_name in natural_sort(trackers):
+            config = self._get_tracker_input_config(tracker_name)
+
+            # Need to handle mpls specifically
+            record_export = config.record_export._cast_as(EosCliConfigGen.FlowTracking.Sampled.TrackersItem.RecordExport)
+            record_export.mpls = config.sampled.record_export.mpls
 
-        return flow_tracking
+            self.structured_config.flow_tracking.sampled.trackers.append_new(
+                name=config.name,
+                record_export=record_export,
+                exporters=config.exporters._cast_as(EosCliConfigGen.FlowTracking.Sampled.TrackersItem.Exporters),
```

Comment:
Could we update the exporters schema to use a pure $ref instead of repeating it from eos_cli_config_gen? Effectively it should be the same, and then the classes would be cli gen classes already.

Reply:
discussed on call we can't because we would need to ref a defs which we can't today